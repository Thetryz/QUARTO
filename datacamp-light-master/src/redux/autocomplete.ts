import { Epic } from "redux-observable";
import { Record } from "immutable";
import { reducerWithInitialState } from "typescript-fsa-reducers";
import actionCreatorFactory, { AnyAction as Action } from "typescript-fsa";
import { some } from "lodash";
import { Observable } from "rxjs/Observable";
import "rxjs/add/observable/empty";
import "rxjs/add/observable/from";
import "rxjs/add/operator/do";
import "rxjs/add/operator/filter";
import "rxjs/add/operator/combineLatest";
import "rxjs/add/operator/concatMapTo";
import "rxjs/add/operator/mergeMap";
import { SessionOutput } from "@datacamp/multiplexer-client";

import { State } from "./index";
import { muxRegistered } from "./backend-session";
import { selectId } from "./exercise";

import { getMux as muxGetter } from "../helpers/pluginManager";

/*
 * Actions */

const createAction = actionCreatorFactory("autocomplete");

export const registerConsoleAutocompleteCallback = createAction<
  (output: string) => void
>("REGISTER_CONSOLE_AUTOCOMPLETE_CALLBACK");

/*
 * State definition */

export interface IAutocompleteState {
  consoleAutocompleteCallback: (output: string) => void;
}

const initialState: IAutocompleteState = {
  consoleAutocompleteCallback: output => null,
};

export class AutocompleteState extends Record(initialState) {}

/*
 * Reducer */

const reducer = reducerWithInitialState(new AutocompleteState())
  .case(registerConsoleAutocompleteCallback, (state, callback) =>
    state.set("consoleAutocompleteCallback", callback)
  )
  .build();

export default reducer;

/*
 * Selectors */

export const selectAutocomplete = (state: State) => state.get("autocomplete");

export const selectConsoleAutocompleteCallback = (state: State) =>
  selectAutocomplete(state).get("consoleAutocompleteCallback");

/*
 * Epics */

export const createEpicRegisterAutocompleteMuxCallbacks = (
  getMux: typeof muxGetter
): Epic<Action, State> => (action$, store) =>
  action$
    .filter(muxRegistered.match)
    .combineLatest(action$.filter(registerConsoleAutocompleteCallback.match))
    .do(() => {
      console.log("registering autocomplete");
      getMux(selectId(store.getState()))
        .output$.flatMap(outputs => Observable.from(outputs))
        .filter(output => output.type === "code-completion")
        .subscribe(output =>
          selectConsoleAutocompleteCallback(store.getState())(output as any)
        );
    })
    .concatMapTo(Observable.empty());

export const epicRegisterAutocompleteMuxCallbacks = createEpicRegisterAutocompleteMuxCallbacks(
  muxGetter
);
