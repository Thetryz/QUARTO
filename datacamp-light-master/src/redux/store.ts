import { Map as hashMap } from "immutable";
import { compose, createStore, GenericStoreEnhancer } from "redux";
import reducer, { State } from "./";

export default (storeEnhancer: GenericStoreEnhancer) => {
  const store = createStore<State>(reducer, new State(), storeEnhancer);

  if (module.hot) {
    // Enable webpack hot module replacement for reducers
    module.hot.accept("./", () => store.replaceReducer(reducer));
  }

  return store;
};
