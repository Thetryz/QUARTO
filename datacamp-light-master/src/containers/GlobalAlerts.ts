import { connect, MapStateToProps } from "react-redux";

import AlertList, { IAlertListProps } from "../components/AlertList";

import { State } from "../redux";
import { selectAlerts } from "../redux/view";

const mapStateToProps: MapStateToProps<IAlertListProps, {}> = (
  state: State
) => ({
  alerts: selectAlerts(state),
});

export default connect(mapStateToProps)(AlertList);
