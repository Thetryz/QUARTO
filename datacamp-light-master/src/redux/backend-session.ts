import { Epic } from "redux-observable";
import { Record } from "immutable";
import { reducerWithInitialState } from "typescript-fsa-reducers";
import actionCreatorFactory, { AnyAction as Action } from "typescript-fsa";
import { toUpper, find, isEmpty } from "lodash";
import * as cookies from "browser-cookies";
import * as store from "store";
import { BehaviorSubject } from "rxjs/BehaviorSubject";
import { Observable } from "rxjs/Observable";
import "rxjs/add/operator/concat";
import "rxjs/add/observable/of";
import "rxjs/add/operator/filter";
import "rxjs/add/operator/map";
import "rxjs/add/observable/empty";
import "rxjs/add/operator/distinctUntilChanged";
import "rxjs/add/operator/first";
import "rxjs/add/observable/merge";
import "rxjs/add/operator/concatMap";
import "rxjs/add/operator/concatMapTo";
import "rxjs/add/operator/mergeMap";
import "rxjs/add/observable/zip";
import "rxjs/add/operator/do";
import "rxjs/add/observable/combineLatest";

import Multiplexer, {
  Command,
  Language,
  PLUGIN_NAME as MUX_NAME,
  SessionOutput,
  SessionStatus,
} from "@datacamp/multiplexer-client";

import { State } from "./";
import { setId } from "./exercise";
import { viewsStart } from "./view";
import { selectId, selectLanguage, selectListener } from "./exercise";

import ExerciseService from "../helpers/baseExercise";
import getOutputAction from "../helpers/output";
import pluginManager, { getMux } from "../helpers/pluginManager";
import uuid from "../helpers/uuid";

import config from "../config";

/*
 * Actions */

const createAction = actionCreatorFactory("backendSession");

export const startSession = createAction<{
  language: Language;
  id: string;
  force_new: boolean;
  initCommands?: Command;
}>("EPIC_START_SESSION");

export interface ISubmitCodePayload {
  code?: string;
  command?: string;
  id?: number;
  language?: string;
  pec?: string;
  sct?: string;
  solution?: string;
  type?: string;
  figureIndex?: number;
  width?: number;
  height?: number;
}

export const submitCode = createAction<ISubmitCodePayload>("EPIC_SUBMIT_CODE");

export const backendError = createAction<{
  output: string;
}>("EPIC_BACKEND_ERROR");

export const muxRegistered = createAction("EPIC_MUX_REGISTERED");

export const updateBackendStatus = createAction<SessionStatus>(
  "UPDATE_BACKEND_STATUS"
);

export const restartSession = createAction("RESTART_SESSION");

/*
 * State definition */

export const BACKEND_STATUS: { [x: string]: SessionStatus } = {
  BROKEN: { status: "broken", text: "Session Broken" },
  BUSY: { status: "busy", text: "Session Busy" },
  NONE: { status: "none", text: "" },
  READY: { status: "ready", text: "Workspace Ready" },
};

export interface IBackendSessionState {
  status: SessionStatus;
  isInitSession: boolean;
  countPostNewSession: number;
  statusCode: number;
}

const initialState: IBackendSessionState = {
  countPostNewSession: 0,
  isInitSession: false,
  status: BACKEND_STATUS.NONE,
  statusCode: 0,
};

export class BackendSessionState extends Record(initialState) {}

/*
 * Reducer */

const setStatus = (prevState: BackendSessionState, status: SessionStatus) => {
  let state = prevState;

  if (
    state.get("status").status === BACKEND_STATUS.NONE.status &&
    status.status === BACKEND_STATUS.BUSY.status
  ) {
    state = state.set("isInitSession", true);
  }

  return state.set("status", status);
};

const reducer = reducerWithInitialState(new BackendSessionState())
  .case(updateBackendStatus, (state, action) => {
    return setStatus(
      state,
      BACKEND_STATUS[toUpper(action.status)] || BACKEND_STATUS.NONE
    );
  })
  .case(backendError, state => setStatus(state, BACKEND_STATUS.BROKEN))
  .case(startSession, state =>
    state.set("countPostNewSession", state.get("countPostNewSession") + 1)
  )
  .build();

export default reducer;

/*
 * Selectors */

export const selectBackendSession = (state: State) =>
  state.get("backendSession");

export const selectBackendSessionStatusCode = (state: State) =>
  selectBackendSession(state).get("status").status;

export const selectBackendSessionIsBroken = (state: State) =>
  selectBackendSession(state).get("status").status === "broken";

export const selectBackendSessionIsBusy = (state: State) =>
  selectBackendSession(state).get("status").status === "busy";

export const selectBackendSessionCountPostNewSession = (state: State) =>
  selectBackendSession(state).get("countPostNewSession");

export const selectBackendSessionStatus = (state: State) =>
  selectBackendSession(state).get("status");

export const selectBackendUIStatus = (state: State) => {
  const status = selectBackendSession(state).get("status");
  return {
    code: status.status,
    text: status.text,
  };
};

/*
 * Epics */

export const guardOnBackendError = (
  data: SessionOutput[],
  language: Language
) => {
  const backendErrorType = (() => {
    switch (language) {
      case "revo":
      case "r":
        return "error";
      default:
        return "backend-error";
    }
  })();
  const error = find(data, output => output.type === backendErrorType);
  if (error) {
    return Observable.of(backendError({ output: error.payload })); // TODO What's the type?
  }
  return Observable.empty();
};

export const handleInputResponse = (
  data: SessionOutput[],
  language: Language
) => {
  let obs = Observable.of();
  if (!isEmpty(data)) {
    const error$ = guardOnBackendError(data, language);
    obs = obs.concat(error$);
  }
  return obs;
};

const registerMux = (language: Language, dclId: string) => {
  let id = store.get("session-id");
  if (!id) {
    id = uuid();
    store.set("session-id", id);
  }
  const userInfo = {
    authentication_token: id,
    email: `datacamp-light+${id}@datacamp.com`,
  };
  pluginManager.remove(MUX_NAME + dclId);
  pluginManager.register(MUX_NAME + dclId, Multiplexer.AsyncSession, userInfo, {
    debug: process.env.NODE_ENV === "development",
    flattenOutputs: false,
    language,
    multiplexerUrl: config.urls.multiplexer,
  });
};

export const id$ = (action$: Observable<Action>) =>
  action$
    .filter(setId.match)
    .distinctUntilChanged((a, b) => a.payload === b.payload)
    .map(action => action.payload);

export const readyAgain$ = (action$: Observable<Action>) =>
  action$
    .filter(updateBackendStatus.match)
    .distinctUntilChanged((a, b) => a.payload.status === b.payload.status)
    .filter(action => action.payload.status === BACKEND_STATUS.READY.status)
    .map(action => action.payload);

export const epicRegisterMux: Epic<Action, State> = (action$, store) =>
  action$
    .filter(viewsStart.match)
    .first()
    .mergeMap(() => {
      const state = store.getState();
      const dclId = selectId(store.getState());
      const language = selectLanguage(store.getState());
      registerMux(language, dclId);

      const muxStateSubject$ = new BehaviorSubject(BACKEND_STATUS.NONE);
      const updateBackendStatus$ = muxStateSubject$.map(status =>
        updateBackendStatus(status)
      );

      getMux(dclId).subscribe(status => muxStateSubject$.next(status));

      return Observable.merge(
        Observable.of(muxRegistered()),
        Observable.of(
          startSession({
            force_new: false,
            id: dclId,
            initCommands: {
              ...ExerciseService.prepareSubmit(state, {}),
              command: "init",
            },
            language,
          })
        ),
        updateBackendStatus$,
        getMux(dclId).output$.concatMap(
          data => handleInputResponse(data, language) as Observable<Action>
        )
      );
    });

export const epicStartSession: Epic<Action, State> = (action$, store) => {
  return Observable.combineLatest(
    action$.filter(startSession.match),
    action$.filter(setId.match)
  )
    .do(([initializingAction, setIdAction]) => {
      if (initializingAction.payload.force_new) {
        cookies.erase("mux-session-id");
        cookies.erase("AWSELB");
      }
    })
    .do(([startSessionAction, setIdAction]) => {
      const dclId = selectId(store.getState());
      const mux = getMux(dclId);
      const forceNew =
        startSessionAction.payload.force_new ||
        selectBackendSessionCountPostNewSession(store.getState()) === 1;
      const options = {
        language: startSessionAction.payload.language,
        force_new: forceNew,
      };

      mux.start(options, startSessionAction.payload.initCommands);
    })
    .do(() => selectListener(store.getState())("start", undefined))
    .concatMapTo(Observable.empty());
};

export const epicSubmitCode: Epic<Action, State> = (action$, store) => {
  return Observable.combineLatest(
    id$(action$),
    Observable.zip(
      readyAgain$(action$),
      action$.filter(submitCode.match),
      (backendStatus, submitAction) => ({ backendStatus, submitAction })
    )
  )
    .map(([dclId, { submitAction }]) => {
      const state = store.getState();
      const commandConfig = ExerciseService.prepareSubmit(
        state,
        submitAction.payload
      );

      getMux(dclId).input({
        ...commandConfig,
      });

      return commandConfig;
    })
    .do(config =>
      selectListener(store.getState())("submit", {
        code: config.code,
      })
    )
    .concatMapTo(Observable.empty());
};

export const epicRestartSession: Epic<Action, State> = (action$, store) =>
  action$.filter(restartSession.match).map(() =>
    startSession({
      language: selectLanguage(store.getState()),
      id: selectId(store.getState()),
      force_new: true,
    })
  );
