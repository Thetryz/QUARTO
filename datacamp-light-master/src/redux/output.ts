import { Epic } from "redux-observable";
import { Record, List } from "immutable";
import { reducerWithInitialState } from "typescript-fsa-reducers";
import actionCreatorFactory, { AnyAction as Action } from "typescript-fsa";
import { Observable } from "rxjs/Observable";
import "rxjs/add/observable/empty";
import "rxjs/add/observable/from";
import "rxjs/add/observable/zip";
import "rxjs/add/operator/combineLatest";
import "rxjs/add/operator/concatMapTo";
import "rxjs/add/operator/do";
import "rxjs/add/operator/filter";
import "rxjs/add/operator/map";
import "rxjs/add/operator/mapTo";
import "rxjs/add/operator/mergeMap";
import "rxjs/add/operator/switchMap";
import "rxjs/add/operator/ignoreElements";
import "rxjs/add/operator/take";
import { includes, some } from "lodash";
import { SessionOutput } from "@datacamp/multiplexer-client";
import { IPlotSource } from "@datacamp/ui-plot";

import { State } from "./";
import {
  selectId,
  selectLanguage,
  selectType,
  selectExercise,
} from "./exercise";
import { muxRegistered, submitCode } from "./backend-session";
import { expandTab, setRenderSize } from "./view";

import getOutputAction, { outputsForConsole } from "../helpers/output";
import { getMux } from "../helpers/pluginManager";
import { expandHandler } from "../helpers/plot";

/*
 * Actions */

const createAction = actionCreatorFactory("output");

export const registerConsoleOutputCallback = createAction<
  (result: SessionOutput[]) => void
>("REGISTER_CONSOLE_OUTPUT_CALLBACK");

export const addPlot = createAction<IPlotSource>("ADD_PLOT");

export const setPlotIndex = createAction<number>("SET_PLOT_INDEX");

export const setPlot = createAction<{
  figureIndex: number;
  source: IPlotSource;
}>("SET_PLOT");

/*
 * State definition */

export interface IOutputState {
  consoleOutputCallback: (output: SessionOutput[]) => void;
  plots: List<IPlotSource>;
  plotIndex: number;
}

const initialState: IOutputState = {
  consoleOutputCallback: output => null,
  plots: List<IPlotSource>(),
  plotIndex: 0,
};

export class OutputState extends Record(initialState) {}

/*
 * Reducer */

export const reducer = reducerWithInitialState(new OutputState())
  .case(registerConsoleOutputCallback, (state, callback) =>
    state.set("consoleOutputCallback", callback)
  )
  .case(addPlot, (state, source) =>
    state
      .update("plots", plots => plots.push(source))
      .update("plotIndex", _ => Math.max(0, state.get("plots").size))
  )
  .case(setPlotIndex, (state, index) => state.set("plotIndex", index))
  .case(setPlot, (state, { figureIndex, source }) =>
    state.update("plots", plots => plots.set(figureIndex, source))
  )
  .build();

export default reducer;

/*
 * Selectors */

export const selectOutput = (state: State) => state.get("output");

export const selectConsoleOutputCallback = (state: State) =>
  selectOutput(state).get("consoleOutputCallback");

export const selectPlots = (state: State) =>
  selectOutput(state)
    .get("plots")
    .toJS();

export const selectPlotIndex = (state: State) =>
  selectOutput(state).get("plotIndex");

export const selectPlotSource = (state: State) =>
  selectOutput(state)
    .get("plots")
    .toJS()[selectPlotIndex(state)];

/*
 * Epics */

export const epicRegisterOutputMuxCallbacks: Epic<Action, State> = (
  action$,
  store
) =>
  action$
    .filter(muxRegistered.match)
    .combineLatest(action$.filter(registerConsoleOutputCallback.match))
    .do(() =>
      getMux(selectId(store.getState())).subscribeToOutputs(
        (output: SessionOutput) => includes(outputsForConsole, output.type),
        selectConsoleOutputCallback(store.getState())
      )
    )
    .concatMapTo(Observable.empty());

export const epicOutputFeedback: Epic<Action, State> = (action$, store) =>
  action$
    .filter(muxRegistered.match)
    .flatMap(
      () =>
        getMux(selectId(store.getState()))
          .output$.flatMap(outputs => Observable.from(outputs))
          .map(output => {
            const state = store.getState();
            return getOutputAction(
              selectLanguage(state),
              output,
              selectType(state)
            );
          }) as Observable<Action>
    )
    .filter(action => action !== null);
