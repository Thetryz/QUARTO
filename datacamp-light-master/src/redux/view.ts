import { Epic } from "redux-observable";
import { Record } from "immutable";
import { reducerWithInitialState } from "typescript-fsa-reducers";
import actionCreatorFactory, { AnyAction as Action } from "typescript-fsa";
import { Observable } from "rxjs/Observable";
import "rxjs/add/observable/empty";
import "rxjs/add/observable/of";
import "rxjs/add/operator/filter";
import "rxjs/add/operator/map";
import "rxjs/add/operator/switchMap";
import "rxjs/add/operator/mergeMap";

import { State } from "./";
import { submitCode } from "./backend-session";
import { setPlot, selectPlotIndex, selectPlots } from "./output";
import { ExerciseState, selectLanguage, selectExercise } from "./exercise";

import { IPlotSource } from "@datacamp/ui-plot";
import { expandHandler } from "../helpers/plot";

/*
 * Actions */

const createAction = actionCreatorFactory("view");

export const viewsStart = createAction("VIEWS_START");

export const expandTab = createAction<{
  category: string;
  tabKey: string;
  src: IPlotSource;
}>("EXPAND_TAB");

export const setRenderSize = createAction<{
  width: number;
  height: number;
}>("SET_RENDER_SIZE");

export const setActiveEditor = createAction<string>("SET_ACTIVE_EDITOR");

/*
 * State definition */

export interface IAlert {
  type: "success" | "danger" | "info";
  content: string;
}

export interface IViewState {
  alerts: IAlert[];
  width?: number;
  height?: number;
  activeEditor?: string;
}

const initialState: IViewState = {
  alerts: [],
  width: 640,
  height: 640,
  activeEditor: "editor",
};

export class ViewState extends Record(initialState) {}

/*
 * Reducer */

export const reducer = reducerWithInitialState(new ViewState())
  .case(setRenderSize, (state, action) => state.merge(action))
  .case(setActiveEditor, (state, editor) => state.set("activeEditor", editor))
  .build();

export default reducer;

/*
 * Selectors */

export const selectView = (state: State) => state.get("view");

export const selectAlerts = (state: State) => selectView(state).get("alerts");

export const selectRenderWidth = (state: State) =>
  selectView(state).get("width");

export const selectRenderHeight = (state: State) =>
  selectView(state).get("height");

export const selectActiveEditor = (state: State) =>
  selectView(state).get("activeEditor");

/*
 * Epics */

export const epicResizePlot: Epic<Action, State> = (action$, store) =>
  action$.filter(setRenderSize.match).mergeMap(action => {
    const state = store.getState();
    const language = selectLanguage(state);
    const figureIndex = selectPlotIndex(state);
    const plots = selectPlots(state);

    if (plots.length <= 0) {
      return Observable.empty();
    }

    if (language === "r" || language === "revo") {
      return Observable.of(
        submitCode({
          language: selectLanguage(state),
          command: "resize",
          figureIndex,
          ...action.payload,
        })
      );
    } else {
      return Observable.empty();
    }
  });

export const epicExpandTab: Epic<Action, State> = (action$, store) =>
  action$.filter(expandTab.match).switchMap(action => {
    const state = store.getState();
    const currentExercise = selectExercise(state).toJS() as ExerciseState;
    const figureIndex = selectPlotIndex(state);
    const type = action.payload.tabKey;

    if (currentExercise.language === "r" && action.payload.src.type === "img") {
      return Observable.of(
        submitCode({
          language: currentExercise.language,
          command: "expand",
          width: 800,
          height: 800,
          figureIndex,
        })
      );
    } else {
      expandHandler(
        type,
        selectPlots(state)[figureIndex],
        figureIndex,
        currentExercise
      );
      return Observable.empty();
    }
  });
