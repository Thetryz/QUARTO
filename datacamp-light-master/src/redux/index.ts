import { Action } from "redux";
import { Epic } from "redux-observable";
import * as request from "universal-rx-request";
import { Map as HashMap, Record } from "immutable";
import { combineReducers } from "redux-immutable";

import autocomplete, { AutocompleteState } from "./autocomplete";
import backendSession, { BackendSessionState } from "./backend-session";
import exercise, { ExerciseState } from "./exercise";
import output, { OutputState } from "./output";
import view, { ViewState } from "./view";

/*
 * State definition */

export interface IState {
  autocomplete: AutocompleteState;
  backendSession: BackendSessionState;
  exercise: ExerciseState;
  output: OutputState;
  view: ViewState;
}

const initialState: IState = {
  autocomplete: new AutocompleteState(),
  backendSession: new BackendSessionState(),
  exercise: new ExerciseState(),
  output: new OutputState(),
  view: new ViewState(),
};

export class State extends Record(initialState) {}

/*
 * Reducer */

const reducer = combineReducers<State>({
  autocomplete,
  backendSession,
  exercise,
  output,
  view,
});

export default reducer;

/*
 * Epics */

request.importRxExtensions();

import { epicRegisterAutocompleteMuxCallbacks } from "./autocomplete";
import {
  epicRegisterMux,
  epicStartSession,
  epicSubmitCode,
  epicRestartSession,
} from "./backend-session";
import { epicPublishFeedback } from "./exercise";
import { epicOutputFeedback, epicRegisterOutputMuxCallbacks } from "./output";
import { epicResizePlot, epicExpandTab } from "./view";

export const epics: Array<Epic<Action, State, any>> = [
  epicRegisterAutocompleteMuxCallbacks,
  epicRegisterMux,
  epicStartSession,
  epicRestartSession,
  epicSubmitCode,
  epicRegisterOutputMuxCallbacks,
  epicOutputFeedback,
  epicResizePlot,
  epicExpandTab,
  epicPublishFeedback,
];
