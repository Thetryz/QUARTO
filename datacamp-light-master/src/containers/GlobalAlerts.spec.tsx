import * as React from "react";
import { shallow } from "enzyme";
import { createStore } from "redux";
import { Provider } from "react-redux";

import reducer, { State } from "../redux";

import GlobalAlerts from "./GlobalAlerts";

describe("GlobalAlerts", () => {
  const store = createStore(reducer, new State());

  it("should exist", () => {
    expect(GlobalAlerts).toBeDefined();
  });

  it("should match snapshot", () => {
    const tree = shallow(
      <Provider store={store}>
        <GlobalAlerts />
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
});
