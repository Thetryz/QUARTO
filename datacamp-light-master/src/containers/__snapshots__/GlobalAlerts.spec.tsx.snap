// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalAlerts should match snapshot 1`] = `<Connect(AlertList) />`;
